pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/impl/json/MutableJsonIndexImpl.java

```diff
@@ -291,6 +297,88 @@ private RoaringBitmap getMatchingFlattenedDocIds(Predicate predicate) {
       } else {
         return new RoaringBitmap();
       }
+    } else if (predicateType == Predicate.Type.REGEXP_LIKE) {
+      String ceilingKey = _postingListMap.ceilingKey(key + BaseJsonIndexCreator.KEY_VALUE_SEPARATOR);
```

Comment:
Do we post key along? If so, we can directly try to find the key, and when calculating the subMap, exclude the lower bound

Reply:
I was suggesting to handle it similar to the immutable one, where we lookup the `key` from the map (`_postingListMap.get(key)`), if it returns `null` it means the key does not exist; if it exists, we can use `key` as lower bound to calculate the `subMap`, and set the lower bound to be exclusive